File: scripts/seed.ts
/**
 * 演示数据 Seed 脚本 V2
 * - 新增 visitors 字段
 * - 推广固定 7 项：货品全站推广/关键词推广/人群推广/店铺直达/内容营销/淘宝客/其它
 * - 月度成本 12 项明细
 * - 生成 90 天每日数据 + 4 个月度成本（覆盖 90 天所在月份）
 */
import { db } from "../src/lib/db";

// 固定 7 个推广渠道
const PROMOTION_FIELDS = [
  "货品全站推广",
  "关键词推广",
  "人群推广",
  "店铺直达",
  "内容营销",
  "淘宝客",
  "其它",
];

const DEMO_STORES = [
  { name: "潮流数码旗舰店", platform: "taobao", shopUrl: "https://shop12345.taobao.com", contact: "张老板", note: "主营数码配件、智能家居" },
  { name: "美妆精选天猫店", platform: "tmall", shopUrl: "https://meizhuang.tmall.com", contact: "李老板", note: "美妆个护品类" },
  { name: "潮流服饰抖店", platform: "douyin", shopUrl: "https://douyin.com/shop/fushi", contact: "王老板", note: "女装直播带货" },
];

const DEMO_SKUS = [
  { skuCode: "SP-001", skuName: "无线蓝牙耳机 Pro", category: "数码", unitCost: 65, unitPrice: 199 },
  { skuCode: "SP-002", skuName: "便携充电宝 20000mAh", category: "数码", unitCost: 45, unitPrice: 129 },
  { skuCode: "SP-003", skuName: "智能手表运动版", category: "数码", unitCost: 120, unitPrice: 359 },
  { skuCode: "SP-004", skuName: "Type-C 数据线 3米", category: "数码", unitCost: 5, unitPrice: 29 },
  { skuCode: "SP-005", skuName: "便携蓝牙音箱", category: "数码", unitCost: 80, unitPrice: 219 },
  { skuCode: "SP-006", skuName: "手机支架铝合金", category: "数码", unitCost: 12, unitPrice: 49 },
  { skuCode: "SP-007", skuName: "智能台灯护眼", category: "智能家居", unitCost: 55, unitPrice: 159 },
  { skuCode: "SP-008", skuName: "加湿器 4L 大容量", category: "智能家居", unitCost: 38, unitPrice: 109 },
  { skuCode: "SP-009", skuName: "电动牙刷声波", category: "个护", unitCost: 28, unitPrice: 99 },
  { skuCode: "SP-010", skuName: "车载空气净化器", category: "智能家居", unitCost: 75, unitPrice: 229 },
  { skuCode: "SP-011", skuName: "USB 拓展坞 7合1", category: "数码", unitCost: 35, unitPrice: 99 },
  { skuCode: "SP-012", skuName: "无线鼠标静音", category: "数码", unitCost: 18, unitPrice: 59 },
];

let seedState = 42;
function rand() {
  seedState = (seedState * 9301 + 49297) % 233280;
  return seedState / 233280;
}
const randRange = (min: number, max: number) => min + rand() * (max - min);
const randInt = (min: number, max: number) => Math.floor(randRange(min, max + 1));
function pick<T>(arr: T[], n: number): T[] {
  const copy = [...arr];
  const result: T[] = [];
  for (let i = 0; i < n && copy.length; i++) {
    const idx = Math.floor(rand() * copy.length);
    result.push(copy.splice(idx, 1)[0]);
  }
  return result;
}

function splitPromotion(total: number): Record<string, number> {
  const weights = PROMOTION_FIELDS.map(() => randRange(0.5, 1.5));
  const sum = weights.reduce((a, b) => a + b, 0);
  const result: Record<string, number> = {};
  PROMOTION_FIELDS.forEach((f, i) => {
    result[f] = Math.round((total * weights[i] / sum) * 100) / 100;
  });
  return result;
}

async function main() {
  console.log("开始 seed 演示数据 V2...");

  const existing = await db.store.count();
  if (existing > 0) {
    console.log(`数据库已有 ${existing} 个店铺，跳过 seed`);
    return;
  }

  // 0. 创建默认用户（demo@ecom.com / demo123）
  const crypto = await import("crypto");
  const salt = "ecom-cockpit-pro-2026";
  const passwordHash = crypto.createHash("sha256").update(salt + "demo123").digest("hex");
  const defaultUser = await db.user.create({
    data: {
      email: "demo@ecom.com",
      passwordHash,
      name: "演示用户",
    },
  });
  console.log(`✓ 创建默认用户: demo@ecom.com (密码: demo123)`);

  // 1. 创建店铺（关联到默认用户）
  for (const s of DEMO_STORES) {
    await db.store.create({ data: { ...s, userId: defaultUser.id } });
  }
  const stores = await db.store.findMany();
  console.log(`✓ 创建 ${stores.length} 个店铺`);

  // 2. SKU
  for (const store of stores) {
    for (const sku of DEMO_SKUS) {
      await db.sku.create({
        data: { ...sku, storeId: store.id, stock: randInt(50, 500) },
      });
    }
  }
  console.log(`✓ 创建 ${DEMO_SKUS.length * stores.length} 个 SKU`);

  // 3. 90 天每日数据
  const today = new Date();
  today.setHours(0, 0, 0, 0);

  // 收集需要生成月度成本的月份
  const monthsToGenerate = new Set<string>();

  for (const store of stores) {
    const skus = await db.sku.findMany({ where: { storeId: store.id } });
    const scale = store.platform === "taobao" ? 1.0 : (store.platform === "tmall" ? 0.7 : 0.5);

    for (let i = 89; i >= 0; i--) {
      const recordDate = new Date(today);
      recordDate.setDate(recordDate.getDate() - i);

      const weekday = recordDate.getDay();
      const weekendBoost = weekday >= 5 ? 1.3 : 1.0;
      const monthEndBoost = recordDate.getDate() >= 28 ? 1.4 : 1.0;
      const monthIdx = Math.floor((89 - i) / 30);
      const trendBoost = 1.0 + monthIdx * 0.05;
      const noise = randRange(0.85, 1.15);

      const salesAmount = Math.round(8000 * scale * weekendBoost * monthEndBoost * trendBoost * noise * 100) / 100;
      const avgPrice = 130;
      const orderCount = Math.max(1, Math.floor(salesAmount / avgPrice));
      // 访客数：转化率约 1.5%-3%
      const conversionRate = randRange(0.015, 0.03);
      const visitors = Math.max(orderCount, Math.floor(orderCount / conversionRate));
      const refundRate = randRange(0.03, 0.09);
      const refundAmount = Math.round(salesAmount * refundRate * 100) / 100;

      // 推广 7 项
      const promoTotalRate = randRange(0.15, 0.25);
      const promoTotal = Math.round(salesAmount * promoTotalRate * 100) / 100;
      const promoData = splitPromotion(promoTotal);

      // 自动计算
      const netSales = Math.round((salesAmount - refundAmount) * 100) / 100;
      const refundRateCalculated = salesAmount > 0 ? Math.round(refundAmount / salesAmount * 10000) / 10000 : 0;
      const promotionRate = salesAmount > 0 ? Math.round(promoTotal / salesAmount * 10000) / 10000 : 0;
      const roi = promoTotal > 0 ? Math.round(salesAmount / promoTotal * 100) / 100 : 0;
      const avgOrderValue = orderCount > 0 ? Math.round(salesAmount / orderCount * 100) / 100 : 0;
      const conversionRateCalculated = visitors > 0 ? Math.round(orderCount / visitors * 10000) / 10000 : 0;

      const record = await db.dailyRecord.create({
        data: {
          storeId: store.id,
          recordDate,
          salesAmount,
          orderCount,
          refundAmount,
          visitors,
          promotionData: JSON.stringify(promoData),
          promotionTotal: promoTotal,
          netSales,
          refundRate: refundRateCalculated,
          promotionRate,
          roi,
          avgOrderValue,
          conversionRate: conversionRateCalculated,
        },
      });

      // 收集月份
      const y = recordDate.getFullYear();
      const m = recordDate.getMonth() + 1;
      monthsToGenerate.add(`${y}-${m}`);

      // SKU 销售数据
      const activeCount = randInt(4, 6);
      const activeSkus = pick(skus, Math.min(activeCount, skus.length));
      const weights = activeSkus.length > 2
        ? [0.4, 0.3, ...Array(activeSkus.length - 2).fill(0.3 / (activeSkus.length - 2))]
        : Array(activeSkus.length).fill(0.5);

      for (let j = 0; j < activeSkus.length; j++) {
        const sku = activeSkus[j];
        const w = weights[j] || 0.1;
        const skuSales = Math.round(salesAmount * w * randRange(0.9, 1.1) * 100) / 100;
        const skuOrders = Math.max(1, Math.floor(orderCount * w));
        const skuQty = Math.max(1, Math.floor(skuOrders * randRange(1.0, 1.5)));
        const skuCost = Math.round(sku.unitCost * skuQty * 100) / 100;
        const skuRefund = Math.round(skuSales * refundRate * 100) / 100;
        const skuGross = Math.round((skuSales - skuCost - skuRefund) * 100) / 100;

        await db.dailySku.create({
          data: {
            dailyRecordId: record.id,
            skuId: sku.id,
            storeId: store.id,
            recordDate,
            salesAmount: skuSales,
            orderCount: skuOrders,
            refundAmount: skuRefund,
            quantity: skuQty,
            cost: skuCost,
            grossProfit: skuGross,
            roi: skuCost > 0 ? Math.round(skuSales / skuCost * 100) / 100 : 0,
            refundRate: Math.round(refundRate * 10000) / 10000,
          },
        });
      }
    }
  }
  console.log(`✓ 创建 90 天每日数据（含访客数、新推广字段）`);

  // 4. 月度成本（为每个有数据的月份生成）
  for (const store of stores) {
    const scale = store.platform === "taobao" ? 1.0 : (store.platform === "tmall" ? 0.7 : 0.5);
    for (const monthKey of monthsToGenerate) {
      const [y, m] = monthKey.split("-").map(Number);
      // 估算当月销售额（按 30 天 × 日均）
      const monthlySales = 8000 * scale * 30;
      const goodsCost = Math.round(monthlySales * 0.45 * 100) / 100;
      const redPacket = Math.round(monthlySales * 0.02 * 100) / 100;
      const labor = Math.round(9000 * scale * 100) / 100;
      const other = Math.round(monthlySales * 0.01 * 100) / 100;
      const consumerExperience = Math.round(monthlySales * 0.005 * 100) / 100;
      const bnplTechFee = Math.round(monthlySales * 0.003 * 100) / 100;
      const basicSoftwareFee = Math.round(monthlySales * 0.004 * 100) / 100;
      const redPacketAdvance = Math.round(monthlySales * 0.002 * 100) / 100;
      const logistics = Math.round(monthlySales * 0.015 * 100) / 100;
      const brandGiftFee = Math.round(monthlySales * 0.002 * 100) / 100;
      const charity = Math.round(monthlySales * 0.001 * 100) / 100;
      const quickPaymentFee = Math.round(monthlySales * 0.002 * 100) / 100;
      const marketingPlatform = Math.round(monthlySales * 0.008 * 100) / 100;

      const totalCost = Math.round((goodsCost + redPacket + labor + other + consumerExperience + bnplTechFee + basicSoftwareFee + redPacketAdvance + logistics + brandGiftFee + charity + quickPaymentFee + marketingPlatform) * 100) / 100;

      await db.monthlyCost.create({
        data: {
          storeId: store.id,
          year: y,
          month: m,
          goodsCost,
          redPacket,
          labor,
          other,
          consumerExperience,
          bnplTechFee,
          basicSoftwareFee,
          redPacketAdvance,
          logistics,
          brandGiftFee,
          charity,
          quickPaymentFee,
          marketingPlatform,
          totalCost,
        },
      });
    }
  }
  console.log(`✓ 创建 ${monthsToGenerate.size} 个月的成本数据`);

  // 5. 利润目标
  const mainStore = stores[0];
  const year = today.getFullYear();
  const month = today.getMonth() + 1;
  const quarter = Math.floor((month - 1) / 3) + 1;
  await db.profitTarget.createMany({
    data: [
      { storeId: mainStore.id, targetType: "yearly", targetYear: year, targetAmount: 3_000_000 },
      { storeId: mainStore.id, targetType: "quarterly", targetYear: year, targetQuarter: quarter, targetAmount: 800_000 },
      { storeId: mainStore.id, targetType: "monthly", targetYear: year, targetMonth: month, targetAmount: 280_000 },
    ],
  });
  console.log(`✓ 创建利润目标`);

  // 6. 设置
  await db.setting.createMany({
    data: [
      { key: "theme", value: "light", description: "主题" },
      { key: "ai_model", value: "glm-4-plus", description: "AI 模型" },
      { key: "auto_backup", value: "weekly", description: "自动备份频率" },
      { key: "currency", value: "CNY", description: "货币" },
      { key: "company_name", value: "我的电商公司", description: "公司名称" },
    ],
  });

  // 7. 预警
  const now = new Date();
  await db.alert.createMany({
    data: [
      {
        storeId: mainStore.id, alertType: "refund_rate_high", level: "warning",
        title: "退款率连续 3 天超过阈值",
        content: `店铺【${mainStore.name}】退款率连续 3 天超过 8%，请关注产品质量和售后服务。`,
        triggeredAt: new Date(now.getTime() - 2 * 24 * 60 * 60 * 1000),
      },
      {
        storeId: mainStore.id, alertType: "promotion_roi_low", level: "critical",
        title: "推广投产比偏低",
        content: "推广投产比低于 4，建议优化关键词和出价策略。",
        triggeredAt: new Date(now.getTime() - 1 * 24 * 60 * 60 * 1000),
      },
    ],
  });
  console.log(`✓ 创建预警数据`);

  console.log("\n🎉 演示数据 V2 写入完成！");
}

main()
  .catch((e) => { console.error("Seed 失败:", e); process.exit(1); })
  .finally(async () => { await db.$disconnect(); });

File: ecom-cockpit-pro/app/database/seed.py
"""
演示数据生成器
=================
首次启动时自动写入：
- 1 个淘宝演示店铺 + 1 个天猫店 + 1 个抖店
- 12 个 SKU
- 90 天每日经营数据（含推广/成本/SKU 销售）
- 年度/季度/月度利润目标
- 历史预警记录
"""
import random
from datetime import date, timedelta, datetime
from typing import List

from sqlalchemy.orm import Session

from app.database.models import (
    Store, SKU, DailyRecord, DailySKU,
    ProfitTarget, Alert, Setting
)
from app.config import PLATFORM_PROMOTION_FIELDS


# 固定随机种子，保证每次生成数据一致
random.seed(42)


# ============== 演示店铺 ==============
DEMO_STORES = [
    {
        "name": "潮流数码旗舰店",
        "platform": "taobao",
        "shop_url": "https://shop12345.taobao.com",
        "contact": "张老板",
        "note": "主营数码配件、智能家居"
    },
    {
        "name": "美妆精选天猫店",
        "platform": "tmall",
        "shop_url": "https://meizhuang.tmall.com",
        "contact": "李老板",
        "note": "美妆个护品类"
    },
    {
        "name": "潮流服饰抖店",
        "platform": "douyin",
        "shop_url": "https://douyin.com/shop/fushi",
        "contact": "王老板",
        "note": "女装直播带货"
    },
]


# ============== 演示 SKU ==============
DEMO_SKUS = [
    {"sku_code": "SP-001", "sku_name": "无线蓝牙耳机 Pro", "category": "数码", "unit_cost": 65, "unit_price": 199},
    {"sku_code": "SP-002", "sku_name": "便携充电宝 20000mAh", "category": "数码", "unit_cost": 45, "unit_price": 129},
    {"sku_code": "SP-003", "sku_name": "智能手表运动版", "category": "数码", "unit_cost": 120, "unit_price": 359},
    {"sku_code": "SP-004", "sku_name": "Type-C 数据线 3米", "category": "数码", "unit_cost": 5, "unit_price": 29},
    {"sku_code": "SP-005", "sku_name": "便携蓝牙音箱", "category": "数码", "unit_cost": 80, "unit_price": 219},
    {"sku_code": "SP-006", "sku_name": "手机支架铝合金", "category": "数码", "unit_cost": 12, "unit_price": 49},
    {"sku_code": "SP-007", "sku_name": "智能台灯护眼", "category": "智能家居", "unit_cost": 55, "unit_price": 159},
    {"sku_code": "SP-008", "sku_name": "加湿器 4L 大容量", "category": "智能家居", "unit_cost": 38, "unit_price": 109},
    {"sku_code": "SP-009", "sku_name": "电动牙刷声波", "category": "个护", "unit_cost": 28, "unit_price": 99},
    {"sku_code": "SP-010", "sku_name": "车载空气净化器", "category": "智能家居", "unit_cost": 75, "unit_price": 229},
    {"sku_code": "SP-011", "sku_name": "USB 拓展坞 7合1", "category": "数码", "unit_cost": 35, "unit_price": 99},
    {"sku_code": "SP-012", "sku_name": "无线鼠标静音", "category": "数码", "unit_cost": 18, "unit_price": 59},
]


def seed_demo_data_if_empty(session: Session):
    """如果数据库为空，写入演示数据"""
    if session.query(Store).count() > 0:
        return  # 已有数据，跳过

    # 1. 创建店铺
    stores = []
    for store_data in DEMO_STORES:
        store = Store(**store_data, is_active=True)
        session.add(store)
        stores.append(store)
    session.flush()

    # 2. 为每个店铺创建 SKU（同样的 12 个 SKU）
    for store in stores:
        for sku_data in DEMO_SKUS:
            sku = SKU(
                store_id=store.id,
                stock=random.randint(50, 500),
                **sku_data
            )
            session.add(sku)
    session.flush()

    # 3. 生成 90 天历史数据
    today = date.today()
    start_date = today - timedelta(days=89)

    for store in stores:
        # 加载该店铺所有 SKU
        skus = session.query(SKU).filter(SKU.store_id == store.id).all()
        if not skus:
            continue

        platform_fields = PLATFORM_PROMOTION_FIELDS.get(store.platform, ["其他"])
        # 平台系数（不同店铺规模不同）
        scale = 1.0 if store.platform == "taobao" else (0.7 if store.platform == "tmall" else 0.5)

        # 90 天循环生成
        for i in range(90):
            record_date = start_date + timedelta(days=i)

            # 周末销售通常更高
            weekday = record_date.weekday()
            weekend_boost = 1.3 if weekday >= 5 else 1.0

            # 月末促销峰
            month_end_boost = 1.4 if record_date.day >= 28 else 1.0

            # 整体增长趋势（每月递增 5%）
            month_idx = i // 30
            trend_boost = 1.0 + month_idx * 0.05

            # 随机波动 ±15%
            noise = random.uniform(0.85, 1.15)

            # 基础销售额
            base_sales = 8000 * scale * weekend_boost * month_end_boost * trend_boost * noise
            sales_amount = round(base_sales, 2)

            # 订单数（基于客单价反推）
            avg_price = 130
            order_count = max(1, int(sales_amount / avg_price))

            # 退款（5%-10% 退款率）
            refund_rate = random.uniform(0.03, 0.09)
            refund_amount = round(sales_amount * refund_rate, 2)
            refund_order_count = int(order_count * refund_rate)

            # 推广数据（占销售额 15%-25%）
            promo_total_rate = random.uniform(0.15, 0.25)
            promo_total = round(sales_amount * promo_total_rate, 2)
            promo_data = _split_promotion(promo_total, platform_fields)

            # 成本数据
            goods_cost_rate = 0.45  # 商品成本占 45%
            goods_cost = round(sales_amount * goods_cost_rate, 2)
            shipping_cost = round(order_count * random.uniform(3, 5), 2)
            package_cost = round(order_count * random.uniform(0.8, 1.5), 2)
            labor_cost = round(300 * scale, 2)  # 人工日固定
            rent_cost = round(200 * scale, 2)   # 房租日固定
            other_cost = round(sales_amount * 0.02, 2)

            cost_data = {
                "商品成本": goods_cost,
                "运费": shipping_cost,
                "包装": package_cost,
                "人工": labor_cost,
                "房租": rent_cost,
                "其他": other_cost,
            }
            cost_total = round(goods_cost + shipping_cost + package_cost + labor_cost + rent_cost + other_cost, 2)

            # 自动计算
            gross_profit = round(sales_amount - goods_cost - refund_amount, 2)
            net_profit = round(gross_profit - promo_total - shipping_cost - package_cost - labor_cost - rent_cost - other_cost, 2)
            profit_rate = round(net_profit / sales_amount, 4) if sales_amount else 0
            roi = round(sales_amount / promo_total, 2) if promo_total else 0
            avg_order_value = round(sales_amount / order_count, 2) if order_count else 0
            profit_per_order = round(net_profit / order_count, 2) if order_count else 0

            record = DailyRecord(
                store_id=store.id,
                record_date=record_date,
                sales_amount=sales_amount,
                order_count=order_count,
                refund_amount=refund_amount,
                refund_order_count=refund_order_count,
                promotion_data=promo_data,
                promotion_total=promo_total,
                cost_data=cost_data,
                cost_total=cost_total,
                gross_profit=gross_profit,
                net_profit=net_profit,
                profit_rate=profit_rate,
                roi=roi,
                avg_order_value=avg_order_value,
                profit_per_order=profit_per_order,
                refund_rate=round(refund_rate, 4),
                promotion_rate=round(promo_total_rate, 4),
            )
            session.add(record)
            session.flush()

            # 为该日生成 SKU 销售数据（销量集中在 3-5 个 SKU 上）
            active_sku_count = random.randint(4, 6)
            active_skus = random.sample(skus, min(active_sku_count, len(skus)))

            # 分配销售额（80% 集中在前 2 个 SKU）
            weights = [0.4, 0.3] + [0.3 / (len(active_skus) - 2)] * (len(active_skus) - 2) if len(active_skus) > 2 else [0.5, 0.5]
            for sku, weight in zip(active_skus, weights):
                sku_sales = round(sales_amount * weight * random.uniform(0.9, 1.1), 2)
                sku_orders = max(1, int(order_count * weight))
                sku_qty = max(1, int(sku_orders * random.uniform(1.0, 1.5)))
                sku_cost = round(sku.unit_cost * sku_qty, 2)
                sku_refund = round(sku_sales * refund_rate, 2)
                sku_gross = round(sku_sales - sku_cost - sku_refund, 2)

                daily_sku = DailySKU(
                    daily_record_id=record.id,
                    sku_id=sku.id,
                    record_date=record_date,
                    store_id=store.id,
                    sales_amount=sku_sales,
                    order_count=sku_orders,
                    refund_amount=sku_refund,
                    refund_order_count=int(sku_orders * refund_rate),
                    quantity=sku_qty,
                    cost=sku_cost,
                    gross_profit=sku_gross,
                    roi=round(sku_sales / max(sku_cost, 0.01), 2),
                    refund_rate=round(refund_rate, 4),
                )
                session.add(daily_sku)

    # 4. 利润目标（淘宝主店）
    main_store = stores[0]
    today = date.today()
    targets = [
        ProfitTarget(store_id=main_store.id, target_type="yearly",
                     target_year=today.year, target_amount=3_000_000),
        ProfitTarget(store_id=main_store.id, target_type="quarterly",
                     target_year=today.year, target_quarter=(today.month - 1) // 3 + 1,
                     target_amount=800_000),
        ProfitTarget(store_id=main_store.id, target_type="monthly",
                     target_year=today.year, target_month=today.month, target_amount=280_000),
    ]
    for t in targets:
        session.add(t)

    # 5. 默认设置
    settings = [
        Setting(key="theme", value="light", description="主题"),
        Setting(key="ai_model", value="glm-4-plus", description="AI 模型"),
        Setting(key="auto_backup", value="weekly", description="自动备份频率"),
        Setting(key="currency", value="CNY", description="货币"),
        Setting(key="company_name", value="我的电商公司", description="公司名称"),
    ]
    for s in settings:
        session.add(s)

    # 6. 历史预警（最近 7 天的几条）
    for store in stores[:1]:  # 只为主店生成
        alerts = [
            Alert(store_id=store.id, alert_type="refund_rate_high",
                  level="warning",
                  title="退款率连续 3 天超过阈值",
                  content=f"店铺【{store.name}】退款率连续 3 天超过 8%，请关注产品质量和售后服务。",
                  triggered_at=datetime.now() - timedelta(days=2)),
            Alert(store_id=store.id, alert_type="promotion_roi_low",
                  level="critical",
                  title="直通车 ROI 偏低",
                  content="直通车 ROI 低于 1.5，建议优化关键词和出价策略。",
                  triggered_at=datetime.now() - timedelta(days=1)),
            Alert(store_id=store.id, alert_type="profit_decline",
                  level="warning",
                  title="昨日净利润环比下降 18%",
                  content="昨日净利润环比下降 18%，主要受推广成本上升影响。",
                  triggered_at=datetime.now() - timedelta(hours=12)),
        ]
        for a in alerts:
            session.add(a)

    session.commit()
    print(f"[Seed] 已写入演示数据：3 个店铺、12 个 SKU、90 天经营数据")


def _split_promotion(total: float, fields: List[str]) -> dict:
    """按平台字段拆分推广预算"""
    if not fields:
        return {"其他": total}
    weights = [random.uniform(0.5, 1.5) for _ in fields]
    total_w = sum(weights)
    return {field: round(total * w / total_w, 2) for field, w in zip(fields, weights)}
